controllers/providers/pxc/provider.go

```diff
@@ -130,23 +131,27 @@ func (p *Provider) Apply(ctx context.Context) everestv1alpha1.Applier {
 // The pause may be overwritten since the Everest operator will also fight for the desired pause state
 // set by the user.
 // To avoid this race condition, we will explicitly inspect every phase of the restore, and set pause accordingly.
-func (p *Provider) handlePauseForPXCRestore(ctx context.Context) error {
-	restores, err := common.ListDatabaseClusterRestores(ctx, p.C, p.DB.GetName(), p.DB.GetNamespace())
-	if err != nil {
-		return err
+func (p *Provider) handlePauseForPXCRestore(ctx context.Context, opts providers.ProviderOptions) error {
+	restores := &pxcv1.PerconaXtraDBClusterRestoreList{}
```

Comment:
Since this is handled in the db-cluster controller, shouldn't it also watch pxc-restores now? Otherwise there could be a delay in this reconciliation..

Reply:
A good point. The db-cluster controller watches `DatabaseClusterRestore`, and `DatabaseClusterRestore` should be reconciled each time the upstream restore object is reconciled... need to take a closer look to it

Reply:
Checked it. There could be a delay indeed if the restoration doesn't go the happy path, so in that case we need to watch the `pxc-restore` resources to be sure the db gets reconciled when the `pxcv1.RestoreStopCluster` status is met. 
The downside is that for pxc we're going to have extra reconciliation calls, bc we watch both `dbr` and `pxc-restore`. I don't really like the solution, but it at least it solves the problems we faced. @mayankshah1607 wdyt?

Reply:
Resolving the thread bc
- the suggestion was implemented
- Mayank is temporary out of work
- need to move forward with the release